docs/source/getting_started/dataset_creation/dataset_creation_gcloud_bucket.rst

```diff
@@ -1,6 +1,164 @@
+configure your dataset to use said bucket, and only upload metadata to Lightly
+while keeping your data private.
+
+
+Setting up Google Cloud Storage
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Lightly needs to be able to create so-called
+`presigned URLs/read URLs <https://cloud.google.com/storage/docs/access-control/signed-urls>`_
+for displaying your data in your browser.
+Thus it needs at minimum to be able to read and list permissions on your bucket.
+If you want Lightly to create optimal thumbnails for you
+while uploading the metadata of your images, write permissions are also needed.
+
+Let us assume the bucket is called `lightly-datalake`.
+And let us assume the folder you want to use with Lightly is located at `projects/wild-animals/`
+
+**Setting Up a Service Account in IAM**
+
+1. Write down your project ID.
+You find it in the `gcloud console <console.cloud.google.com/home/dashboard>`_ under `Project Info`.
+
+2. Navigate to your bucket in the `google cloud storage browser <https://console.cloud.google.com/storage/browser>`_
+and from there to `projects/wild-animals/`. Copy the path, in this case
+`lightly-datalake/projects/wild-animals`.
+
+.. figure:: ./images_gcloud_bucket/screenshot_gcloud_bucket_project.jpg
+    :align: center
+    :alt: Browsing a google cloud storage bucket.
+    :width: 60%
+
+3. Navigate to the tab `Permissions`.
+Make sure that your access control is uniform.
+If it is not, change it to uniform.
+
+.. figure:: ./images_gcloud_bucket/screenshot_gcloud_uniform_access.jpg
+    :align: center
+    :alt: Ensuring a google cloud bucket has uniform access.
+    :width: 60%
+
+4. Navigate to `IAM & Admin -> Roles <https://console.cloud.google.com/iam-admin/roles>`_.
+
+- There create a new role, with the title and ID `STORAGE_READ_WRITE_LIST`.
+- Click on `Add Permissions`, search for `storage.objects`
+- Add the permissions `storage.objects.get`, `storage.objects.list`, and `storage.objects.create`.
+  The create permissions are needed so that Lightly can create thumbnails in your bucket.
+- After adding the permissions, create the role.
+
+.. figure:: ./images_gcloud_bucket/screenshot_gcloud_storage_role.jpg
+    :align: center
+    :alt: Creating a role for accessing google cloud storage.
+    :width: 60%
+
+5. Navigate to `APIs -> Credentials <https://console.cloud.google.com/apis/credentials>`_.
+
+- Click on `Create Credentials`, choose `Service Account` and insert the name
+  `LIGHTLY_USER_WILD_ANIMALS`.
+- The description can be `service account for the Lightly API to access the wild animals dataset`.
+- Click on `Create and Continue`.
+- Choose the Role you just created, i.e. `STORAGE_READ_WRITE_LIST`.
+- Add a condition with the title `BUCKET_PROJECTS_WILD_ANIMALS`
+  and insert the condition below in the Condition editor. Remember to change the bucket name
+  and path to the folder. However, you must keep the "objects" inbetween.
+  For more information, head to the `IAM conditions
```

Comment:
great explanation. I would add it to after the actual code snippet

Reply:
^this

Reply:
I moved it.